skbase/base/_base.py

```diff
@@ -77,6 +77,24 @@ def __init__(self):
         self._tags_dynamic = {}
         super(BaseObject, self).__init__()
 
+    def __eq__(self, other):
+        """Equality dunder. Checks equal class and parameters.
+
+        Returns True iff result of get_params(deep=False)
+        results in equal parameter sets.
+
+        Nested BaseObject descendants from get_params are compared via __eq__ as well.
+        """
+        from skbase.testing.utils.deep_equals import deep_equals
```

Comment:
Is there a use-case where we'd want this logic outside of a BaseObject?

It may make sense to add a public function to skbase.validate that performs this comparison and then import it here.

Reply:
Note that we could also open up an Issue to do that later.

Reply:
well, `deep_equals` is typically used in tests.
In `sktime`, it is not imported outside tests except here.

Reply:
not sure what the best thing to do is. Recommendations?

Reply:
Lets leave as is and open an Issue to consider porting the logic to `skbase.validate` too.